test/test-PerfMeter.js

```diff
@@ -0,0 +1,196 @@
+      }
+    };
+
+    if (notifyEvent) {
+      Services.obs.addObserver(observer, notifyEvent);
+    }
+
+    // open the page in timeout, to make it fire after we return
+    tabs.open({url: openUrl, onPageShow: onShow});
+  });
+}
+
+function prefSetter(value) {
+  return new Promise(resolve => {
+    let observer = {
+      observe() {
+        Services.obs.removeObserver(this, "performance-pref-changed");
+        resolve();
+      }
+    };
+    Services.obs.addObserver(observer, "performance-pref-changed");
+    simplePrefs.prefs.performanceLog = value;
+  });
+}
+
+function verifyPerfEvents(eventsArray, assert) {
+  let exectedEvents = new Set([
+    "TAB_OPEN",
+    "WORKER_ATTACHED",
+    "TAB_READY",
+    "TOP_FRECENT_SITES_REQUEST",
+    "RECENT_LINKS_REQUEST",
+    "FRECENT_LINKS_REQUEST",
+    "RECENT_BOOKMARKS_REQUEST",
+    "SEARCH_STATE_REQUEST",
+    "NOTIFY_PERFORMANCE",
+    "TOP_FRECENT_SITES_RESPONSE",
+    "RECENT_LINKS_RESPONSE",
+    "FRECENT_LINKS_RESPONSE",
+    "RECENT_BOOKMARKS_RESPONSE",
+    "SEARCH_STATE_RESPONSE",
+    "NOTIFY_PERFORMANCE",
+  ]);
+
+  let seenEvents = new Set();
+  for (let event of eventsArray) {
+    assert.ok(exectedEvents.has(event.tag), "Received expected event " + event.tag);
+    assert.ok(!seenEvents.has(event.tag + event.data), "Event tag and data pair is unique");
+    if (event.tag === "NOTIFY_PERFORMANCE") {
+      assert.ok(event.data === "BASE_MOUNTED"  ||
+                event.data === "NEWTAB_RENDER" ||
+                event.data.startsWith("DOC_READY_STATE"),
+                "Correct data associated with content notification");
+    }
+    if (event.tag.indexOf("_RESPONSE") > -1) {
+      // all responses must have delta computed
+      assert.ok(Number.isInteger(event.delta), "Responses have time delta attached");
+    }
+    seenEvents.add(event.tag + event.data);
+  }
+  // the first start must be 0
+  assert.equal(eventsArray[0].start, 0, "First event always start at 0");
+}
+
+exports.test_PerfMeter_init = function(assert) {
+  assert.deepEqual(app.performanceData, {}, "no performance data");
+};
+
+exports.test_PerfMeter_events = function*(assert) {
+  let tabData = yield openTestTab(app.appURLs[1], "performance-log-complete");
+  let notifyData = tabData.notifyData;
+
+  let performanceData = app.performanceData;
+  assert.ok(performanceData[tabData.tab.id], "Entry with expected tab id exists");
+
+  let events = performanceData[tabData.tab.id].events;
+  // did events come as expected
+  verifyPerfEvents(events, assert);
+  // check that "performance-log-complete" notification returns same tabId
+  assert.equal(notifyData.tabId, tabData.tab.id, "Tab ids are identical");
+  // check that "performance-log-complete" notification events are included in application events
+  let eventsSet = new Set(events.map(item => JSON.stringify(item)));
+  notifyData.events.forEach(item => {
+    assert.ok(eventsSet.has(JSON.stringify(item)), "Macthed item " + item.tag);
```

Comment:
Also, why are you comparing JSON stringified items? Unsure the order is guaranteed.